x/nexus/keeper/general_message.go

```diff
@@ -179,17 +187,27 @@ func (k Keeper) getMessages(ctx sdk.Context) (generalMessages []exported.General
 func (k Keeper) GetApprovedMessages(ctx sdk.Context, chain exported.ChainName, limit int64) []exported.GeneralMessage {
 	msgs := []exported.GeneralMessage{}
 	ids := []exported.MessageID{}
-	iter := k.getStore(ctx).IteratorNew(approvedGeneralMessagePrefix.Append(key.From(chain)))
-	defer utils.CloseLogError(iter, k.Logger(ctx))
-	for i := 0; iter.Valid() && i < int(limit); iter.Next() {
+
+	pageRequest := &query.PageRequest{
+		Key:        nil,
+		Offset:     0,
+		Limit:      uint64(limit),
+		CountTotal: false,
+		Reverse:    false,
+	}
+
+	query.Paginate(prefix.NewStore(k.getStore(ctx).KVStore, append(approvedGeneralMessagePrefix.Append(key.From(chain)).Bytes(), []byte(key.DefaultDelimiter)...)), pageRequest, func(key []byte, value []byte) error {
 		var id exported.MessageID
-		iter.UnmarshalValue(&id)
+		k.cdc.MustUnmarshalLengthPrefixed(value, &id)
+
 		ids = append(ids, id)
-		i++
-	}
+		return nil
+	})
+
 	for _, id := range ids {
 		msg, _ := k.GetMessage(ctx, id)
```

Comment:
Use `funcs.MustOk()`. We never want to ignore errors. If the situation isn't expected to occur, then we should panic.